src/sys/event.rs

```diff
@@ -267,8 +267,8 @@ impl KEvent {
     }
 }
 
-pub fn kevent(
-    kq: RawFd,
+pub fn kevent<Fd: AsFd>(
+    kq: &Fd,
```

Comment:
It is better to pass `Fd` as not a reference, consider [`impl<T: AsFd> AsFd for &T`](https://doc.rust-lang.org/nightly/std/os/fd/trait.AsFd.html#impl-AsFd-for-%26T).

Reply:
```rust
use std::{fs::File, os::unix::io::AsFd};

fn syscall<Fd: AsFd>(_fd: Fd) {}

fn main() {
    let f1: File = File::open(".").unwrap();
    let f2: File = f1.try_clone().unwrap();

    syscall(f1);  // We can pass T: AsFd
    syscall(&f2); // and ref to T
}
```
I get it, though all my PRs use `Fd: AsFd(fd: &Fd)` in the current implementations...

cc @asomers, any thoughts on this, which type should we use?

Reply:
Agreed with @JonathanWoollett-Light, see my comment here: https://github.com/nix-rust/nix/issues/1750#issuecomment-1343124228